lib/Graph/Graph.cpp

```diff
@@ -1120,6 +1120,11 @@ LogNode *Function::createLog(llvm::StringRef name, NodeValue input) {
   return addNode(new LogNode(name, input.getType(), input));
 }
 
+LogNode *Function::createLog(llvm::StringRef name, NodeValue input,
```

Comment:
could you merge this and `createLog` above with optional TypeRef outTy arg? That makes Graph interface a bit leaner and does not bring a lot of complexity into createLog impl.